include/boost/graph/isomorphism.hpp

```diff
@@ -158,46 +173,66 @@ namespace detail
             );
         }
 
+        // Generates map of invariant multiplicity from sorted invariants
+        template<typename ForwardIterator>
+        multiplicity_map multiplicities(ForwardIterator first, const ForwardIterator last) 
+        {
+            assert(std::is_sorted(first, last));
+            multiplicity_map invar_multiplicity;
+
+            typedef typename std::vector< invariant_t >::const_iterator invar_iter;
+            typedef typename multiplicity_map::iterator invar_map_iter;
+
+            if(first == last) 
+                return invar_multiplicity;
+            
+            invar_iter it = std::move(first);
```

Comment:
Ahh, not sure what you're thinking here? Just keep using `first`, no need to copy/move it.

Reply:
I always thought range algorithms modifying `first` weren't naming their variables right :man_shrugging: I've removed it.

Reply:
I get what you mean, but I guess one way to think about it is that it's named for what the user passes in. What you do with it inside the algorithm is up to you. And consider, what is the meaning of `it`?

Reply:
That's fair enough, `it` isn't the best variable name either. Could be short for iterator or item, which doesn't help all that much. `iter` might have been better. In any case, I think it's fine the way it is now.